app/parse.py

```diff
@@ -14,5 +17,61 @@ class Course:
     course_type: CourseType
 
 
+COURSE_FIELDS = [field.name for field in fields(Course)]
+
+
+def get_course_type_for_course(course_soup: BeautifulSoup) -> list[CourseType]:
+    course_types_html = (
+        course_soup
+        .find("div", class_="ProfessionCard_buttons__a0o60")
+        .find_all("span", class_="ButtonBody_buttonText__FMZEg")
+    )
+
+    course_type_texts = [
+        course_type.get_text()
+        for course_type in course_types_html
+    ]
+    course_types = []
+    for text in course_type_texts:
+        if text == "Flex":
+            course_types.append(CourseType.PART_TIME)
+        else:
+            course_types.append(CourseType.FULL_TIME)
+    return course_types
+
+
+def parse_single_course(
+        course_soup: BeautifulSoup,
+        course_types: [CourseType]
+) -> list[Course] | Course:
```

Comment:
```suggestion
) -> list[Course]:
```
It always return list, not one instance